Core/src/Vertexing/AdaptiveGridTrackDensity.cpp

```diff
@@ -75,197 +91,171 @@ Acts::AdaptiveGridTrackDensity::getMaxZTPositionAndWidth(
   if (!widthRes.ok()) {
     return widthRes.error();
   }
-  float width = *widthRes;
+  double width = *widthRes;
   ZTPositionAndWidth maxZTAndWidth{maxZT, width};
   return maxZTAndWidth;
 }
 
-typename Acts::AdaptiveGridTrackDensity::DensityMap
-Acts::AdaptiveGridTrackDensity::addTrack(const Acts::BoundTrackParameters& trk,
-                                         DensityMap& mainDensityMap) const {
+AdaptiveGridTrackDensity::DensityMap AdaptiveGridTrackDensity::addTrack(
+    const BoundTrackParameters& trk, DensityMap& mainDensityMap) const {
   ActsVector<3> impactParams = trk.impactParameters();
   ActsSquareMatrix<3> cov = trk.impactParameterCovariance().value();
 
-  // Calculate bin in d direction
-  int centralDBin = getBin(impactParams(0), m_cfg.spatialBinExtent);
-  // Check if current track affects grid density
-  if (std::abs(centralDBin) > (m_cfg.spatialTrkGridSize - 1) / 2.) {
-    DensityMap emptyTrackDensityMap;
-    return emptyTrackDensityMap;
-  }
-  // Calculate bin in z direction
-  int centralZBin = getBin(impactParams(1), m_cfg.spatialBinExtent);
+  int spatialTrkGridSize = getSpatialTrkGridSize(std::sqrt(cov(1, 1)));
+  int temporalTrkGridSize = getTemporalTrkGridSize(std::sqrt(cov(2, 2)));
 
-  // If we don't do time vertex seeding, the time index is set to 0
-  Bin centralBin = std::make_pair(centralZBin, 0.);
+  // Calculate bin in z and t direction
+  int centralZBin = getBin(impactParams(1), m_cfg.spatialBinExtent);
+  int centralTBin = getBin(impactParams(2), m_cfg.temporalBinExtent);
```

Comment:
Hm ok so I see now that you assume that `temporalBinExtent > 0`... Shouldn't we keep the functionality of just doing spatial seeding?

Reply:
Before, if we wanted to do purely spatial seeding, we would set `temporalBinExtent = 0` and then `getBin` would result in a divide by 0 error